docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,355 @@
+```bash
+bigdl-submit \
+    --master yarn \
+    --deploy-mode client \
+    --executor-memory 10g \
+    --driver-memory 10g \
+    --executor-cores 8 \
+    --num-executors 2 \
+    --py-files orca_example.zip \
+    --archives environment.tar.gz#environment \
+    --conf spark.pyspark.driver.python=/path/to/python \
+    --conf spark.pyspark.python=environment/bin/python \
+    train.py
+```
+
+### 3.2.2 Yarn Cluster
+To run every Orca program on yarn cluster with `bigdl-submit`, we need to call `init_orca_context` at the very beginning of the program and specify the `cluster_mode` to `yarn-cluster` to create or get an OrcaContext for yarn-cluster mode.
+
+When running the Fasion-MNIST program on yarn-cluster mode using `bigdl-submit`, we need:
+* Set the `archives` argument to the path of the Conda archive, which will be upload to remote resources(i.e. HDFS) and distributed between executors.
+```bash
+--archives environment.tar.gz#environment
+```
+* For yarn-cluster mode, both driver and executors will use the Python interpreter in `environment.tar.gz`, 
+```bash
+--conf spark.yarn.appMasterEnv.PYSPARK_PYTHON=environment/bin/python \
+--conf spark.executorEnv.PYSPARK_PYTHON=environment/bin/python
+```
+* Set the `py-files` argument to load dependency library to cluster and distribute between executors.
+```bash
+--py-files orca_example.zip
+```
+* For yarn-cluster mode, we recommend that you should use the datasets stored in remote resources like HDFS or S3 instead of downloading to avoid possible network connection errors. You can directly set `remote_dir` argument to the remote datasets path.
+
+When you complete all required preparations, please submit and run the program as below: 
+```bash
+bigdl-submit \
+    --conf spark.yarn.appMasterEnv.PYSPARK_PYTHON=environment/bin/python \
+    --conf spark.executorEnv.PYSPARK_PYTHON=environment/bin/python \
+    --master yarn \
+    --deploy-mode cluster \
+    --executor-memory 10g \
+    --driver-memory 10g \
+    --executor-cores 8 \
+    --num-executors 2 \
+    --archives environment.tar.gz#environment \
+    --py-files orca_example.zip \
+    train.py --remote_dir hdfs://path/to/remote/data
+```
+
+## 3.3 Run Programs with Spark Submit Script
+When the dirver node on the Yarn Cluster is not able to install conda environment, it's recommended for users to use `spark-submit` script to submit and execute the program instead of `bigdl-submit`. Before running with `spark-submit`, we need:
+* Setup spark environment first since we cannot use pyspark on driver, please set environment variables `${SPARK_HOME}` as follows:
+```bash
+export SPARK_HOME=/path/to/spark # the folder path where you extract the Spark package
+```
+* Install all the dependency files that BigDL required (please refer to prepare environment part) on the node which could install conda;
+* Pack the conda environment to an archive on the node with conda then send it to the driver node;
```

Comment:
It's not clear what it means. I think we should install everything on a different node, pack conda env, and then copy to this node?

Reply:
yes, on a different node that could install dependencies using conda.